src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,48 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public abstract class Contact {
+	
+	private final String value;
+	private final boolean isPrivate;
+	
+    /**
+     * Validates given contact using the given regular expression.
+     *
+     * @throws IllegalValueException if given email address string is invalid.
+     */
+	public Contact(String value, boolean isPrivate, String regex, String message_constraints) throws IllegalValueException {
```

Comment:
Interesting design by capturing the validation regex and messages in constructor.